fuse_models/include/fuse_models/common/sensor_proc.h

```diff
@@ -459,7 +471,16 @@ inline bool processDifferentialPoseWithCovariance(
     indices,
     pose_relative_mean_partial,
     pose_relative_covariance_partial);
-  validatePartialMeasurement(pose_relative_mean_partial, pose_relative_covariance_partial);
+
+  try
+  {
+    validatePartialMeasurement(pose_relative_mean_partial, pose_relative_covariance_partial, 1e-6);
+  }
+  catch (const std::runtime_error& ex)
+  {
+    ROS_ERROR_STREAM_THROTTLE(10.0, "Invalid partial differential pose measurement from '"
+                                        << source << "' source: " << ex.what());
+  }
```

Comment:
I feel like if the validation fails, we should not be creating a constraint from it. In the same way that tf errors prevent the creation of a constraint. Thoughts?

```suggestion
    return false;
  }
```

Reply:
The reason I still create constraints it's because I'm playing with https://github.com/clearpathrobotics/fuse/pull/7 and I still get cases where the covariance is slightly non-symmetric or non-PSD, but the constraint with those slightly bad covariances don't make the optimizer explode.

Maybe we could wait til EOW to see if I get things to actually pass this checks. I'm also planning to create an upstream PR for https://github.com/clearpathrobotics/fuse/pull/7 once I get things to work better.

Reply:
Would an extra parameter work? Something like `permit_invalid`?

Reply:
Actually, we could have a parameter to disable the checks. Something like `disable_checks` in Ceres solver.

The checks could be enabled by default, so the user can easily detect issues with the inputs and understand what's wrong with them. This would incur a small overhead.

Once things are known to work and be correct consistently, then `disable_checks: true` could be set, so we don't incur any overhead. If you're ok with something like this, I can add it.

Reply:
@svwilliams What's your opinion on this?

Reply:
I like the "disable_checks" theory. With checks enabled, failed tests prevent constraints from being generated. And with checks disabled, the constraints are obviously generated all the time.

Reply:
Add `disable_checks` param (defaults to `false`) to all sensor models and `validate` argument to all `sensor_proc.h` processing functions.

If `disable_checks == false`, then `validate == true` and all validation checks are evaluated. When a validation check fails, no constraint is created for the partial measurements involved.

If `disable_checks == true`, then `validate == false` and no validation check is evaluated. Therefore, constraints are created for all partial measurements.